File: src/Core/Tests/RazorEngine.Core.VB.Tests/TestTypes/ValueObject.vb
Imports System.Collections.Generic
Imports System.Dynamic
Namespace TestTypes

    Public Class ValueObject
        Inherits DynamicObject
#Region "Fields"
        Private ReadOnly _values As IDictionary(Of String, Object)
#End Region

#Region "Constructor"
        Public Sub New(values As IDictionary(Of String, Object))
            _values = values
        End Sub
#End Region

#Region "Methods"
        Public Overrides Function TryGetMember(binder As GetMemberBinder, ByRef result As Object) As Boolean
            If _values.ContainsKey(binder.Name) Then
                result = _values(binder.Name)
                Return True
            End If

            result = Nothing
            Return False
        End Function
#End Region
    End Class
End Namespace

File: src/Core/Tests/RazorEngine.Core.Tests/TestTypes/ValueObject.cs
namespace RazorEngine.Tests.TestTypes
{
    using System.Collections.Generic;
    using System.Dynamic;

    public class ValueObject : DynamicObject
    {
        #region Fields
        private readonly IDictionary<string, object> _values;
        #endregion

        #region Constructor
        public ValueObject(IDictionary<string, object> values)
        {
            _values = values;
        }
        #endregion

        #region Methods
        public override bool TryGetMember(GetMemberBinder binder, out object result)
        {
            if (_values.ContainsKey(binder.Name))
            {
                result = _values[binder.Name];
                return true;
            }

            result = null;
            return false;
        }
        #endregion
    }
}
